client/coral-admin/src/components/CommentLabels.js

```diff
@@ -6,10 +6,19 @@ import FlagLabel from 'coral-ui/components/FlagLabel';
 import cn from 'classnames';
 import styles from './CommentLabels.css';
 
+const staffRoles = ['ADMIN', 'STAFF', 'MODERATOR'];
+
 function isUserFlagged(actions) {
   return actions.some((action) => action.__typename === 'FlagAction' && action.user);
 }
 
+function getUserFlaggedType(actions) {
+  return actions
+    .filter((action) => action.__typename === 'FlagAction' && action.user)
+    .map((action) => action.user.roles.some((role) => staffRoles.includes(role)))
+    .some((staff) => staff) ? 'Staff' : 'User';
```

Comment:
This chain of methods seems a bit complicated. Why not
```js
function getUserFlaggedType(actions) {
  return actions
    .some((action) =>
      action.__typename === 'FlagAction' &&
      action.user &&
      action.user.roles.some((role) => staffRoles.includes(role))
    ) ? 'Staff' : 'User';
}
```

Reply:
That is cleaner. Updated.